stablehlo/reference/Element.cpp

```diff
@@ -30,15 +30,21 @@ namespace stablehlo {
 
 namespace {
```

Comment:
Thanks for the suggestion!
What is the motivation behind the splitting the `integerFn` into its variants? Is there an op which might need this in future.  Currently we use a common`integerFn` function which handles the variants differently  or uniformly based on the semantics of a particular op.  So far, none of the ops, except for`minimum` and `maximum` need to use different treatment for signed/unsigned variants. The problems with adding the change without many ops using it is that all the ops now has to provide two callbacks which in most of the cases will be the same or throw error for the type being unsupported.

Having said that, `convert` would be a potential op which might need to add additional types in `mapWithUpcastToDouble`.

My proposal is to revisit augmenting these map* functions in the context of an op on a need basis.

Please let me know your opinion.

Reply:
I already have an a pr which uses this, I just split the change.

Reply:
https://github.com/subhankarshah/stablehlo/pull/1

Reply:
I wanted to stack the changes in different prs. This is a pr in my fork.

Reply:
Thanks for sharing. I am afraid that I am still unsure if the handling of  `divOp` is any different from [maxOp](https://github.com/openxla/stablehlo/blob/e28fac48833374bd7686407718953d3801f8dd79/stablehlo/reference/Element.cpp#L293) and we need the split for integer variants. 
But at the same time my call might be biased. 

How about seeking opinion from @burmako  and we can continue with the feedback once we get the green light.